File: DesignHashmap.cs
//used array of linkedlist to create hash map.
//used hash function key%10000; 
//TC: O(1)
//SC:O(N)

public class LinkedListNode {
    public int key;
    public int val;
    public LinkedListNode next;

    public LinkedListNode(int key, int val) {
        this.key = key;
        this.val = val;
        this.next = null;
    }
}

public class MyHashMap {
    private LinkedListNode[] storage;
    private int capacity = 10000;
    public MyHashMap() {
        this.storage = new LinkedListNode[capacity];
    }

    private LinkedListNode find(LinkedListNode head, int key) {
        LinkedListNode prev = head;
        LinkedListNode curr = head.next;
        while (curr != null && curr.key != key) {
            prev = curr;
            curr = curr.next;
        }
        return prev;
    }

    public void Put(int key, int value) {
        int hashfunc1 = key % capacity;
        
        if (storage[hashfunc1] == null) {
            storage[hashfunc1] = new LinkedListNode(-1, -1);  // Sentinel node
        }
        
        LinkedListNode prev = find(storage[hashfunc1], key);
        if (prev.next == null) {
            prev.next = new LinkedListNode(key, value);
        } else {
            prev.next.val = value;
        }
    }

    public int Get(int key) {
        int hashfunc1 = key % capacity;
        if (storage[hashfunc1] == null) {
            return -1;
        }
        
        LinkedListNode prev = find(storage[hashfunc1], key);
        if (prev.next == null) {
            return -1;  // Key not found
        }
        return prev.next.val;
    }

    public void Remove(int key) {
        int hashfunc1 = key % capacity;
        if (storage[hashfunc1] == null) {
            return;  // Key not found
        }
        
        LinkedListNode prev = find(storage[hashfunc1], key);
        if (prev.next == null) {
            return;  // Key not found
        }
        prev.next = prev.next.next;
    }
}

File: design-hashmap.js

var MyHashMap = function() {
    this.buckets = 1000; //using this number since input is betwen 0 and 10^6
    this.map = new Array(this.buckets).fill(null).map(() => []);
};

/** 
 * Hash function to get an index for a key
 * @param {number} key 
 * @return {number}
 */
MyHashMap.prototype.hash = function(key) {
    return key % this.buckets;
}

MyHashMap.prototype.getPrev = function(head, key){
    let prev = null;
    let curr = head;

    while(curr !== null && curr.ket !== key) {
        prev = curr;
        curr = curr.next;
    }
    return prev;
}


/** 
 * @param {number} key 
 * @param {number} value
 * @return {void}
 */
MyHashMap.prototype.put = function(key, value) {
    const idx = this.hash(key);
    const bucket = this.map[idx];

    // Check if the key exists and update it if found
    for (let i = 0; i < bucket.length; i++) {
        if (bucket[i][0] === key) {
            bucket[i][1] = value;
            return;
        }
    }

    // If key doesn't exist, push a new (key, value) pair
    bucket.push([key, value]);
};

/** 
 * @param {number} key
 * @return {number}
 */
MyHashMap.prototype.get = function(key) {
     const idx = this.hash(key);
    const bucket = this.map[idx];

    // Search for the key in the bucket
    for (let i = 0; i < bucket.length; i++) {
        if (bucket[i][0] === key) {
            return bucket[i][1];
        }
    }

    return -1; // Key not found
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashMap.prototype.remove = function(key) {
    const idx = this.hash(key);
    const bucket = this.map[idx];

    // Find the key and remove the entry
    for (let i = 0; i < bucket.length; i++) {
        if (bucket[i][0] === key) {
            bucket.splice(i, 1);
            return;
        }
    }
};

/** 
 * Your MyHashMap object will be instantiated and called as such:
 * var obj = new MyHashMap()
 * obj.put(key,value)
 * var param_2 = obj.get(key)
 * obj.remove(key)
 */